### DISCUSSION TITLE
Suggested best prep for geotiff in GDAL for use in Mapbox GL JS

### DISCUSSION THREAD

**Discussion by gagecarto** [developmentseed/titiler#480] [General]
I've been really enjoying TiTiler and it's performance is quite impressive. I am using it to provide tiles from some COG tiffs.

To prep the geotiffs, I've been warping them to 3857 and then converting to COG like below 

`gdalwarp rasterIn.tif rasterOutCog.tif -of COG`

These are then being loaded on a Mapbox GL JS map. Is there anything else I could do in the gdalwarp step to improve the performance when TiTiler is generating tiles? Peformance is acceptable, I am just wondering if there's any other data prep tricks that would speed thing up.

**Comment by kylebarron** [comment]
If you use `TILING_SCHEME=GoogleMapsCompatible` when creating the cogs, the tiles will internally be aligned with the web mercator grid. And then titiler/rasterio/gdal only needs one get request to get all the data necessary for a tile (along with one request for the cog metadata if it isn't cached). More context in https://gdal.org/drivers/raster/cog.html, https://github.com/developmentseed/titiler/issues/431, https://cogeotiff.github.io/rio-cogeo/Advanced/#web-optimized-cog

**Comment by mradamcox** [comment]
For future reference, I just tried this out with the Python client and it works like so:
```
opts = gdal.TranslateOptions(
    format="COG",
    creationOptions=[
        'COMPRESS=JPEG',
        'TILING_SCHEME=GoogleMapsCompatible',
    ]
)
gdal.Translate(dst_path, src_path, options=opts)
```
and adds this nice bit to `gdalinfo` output:
```
Tiling Scheme:
  NAME=GoogleMapsCompatible
  ZOOM_LEVEL=21
```
Always nice to find little extra things to improve performance...
